ticketbuyer/purchase.go

```diff
@@ -81,11 +82,16 @@ out:
 				case <-quit:
 					return
 				}
-				// Purchase tickets for each attached block, not just for the
-				// update to the main chain.  This is probably not optimal but
-				// it matches how dcrticketbuyer worked.
-				for h := v.NewHeight - int32(len(v.AttachedBlocks)) + 1; h <= v.NewHeight; h++ {
-					p.purchase(int64(h))
+
+				blockHash := v.AttachedBlocks[len(v.AttachedBlocks)-1]
+				blockInfo, err := p.w.BlockInfo(wallet.NewBlockIdentifierFromHash(blockHash))
+				if err != nil {
+					log.Errorf("failed to get block info using block hash %s", blockHash.String())
+				}
+
+				// only try buying tickets on blocks 5 minutes old or less
+				if time.Now().Unix()-blockInfo.Timestamp <= 300 {
```

Comment:
this panics if BlockInfo errored

Reply:
Also ideally instead of performing this check before every single ticket purchase, it should keep checking until it finds a block within the 5 minute interval, and then not do the check again.  after that point it believes it is up to date and should consider buying blocks every new block

Reply:
I think the check should be maintained because for intermittent connections multiple outages can occur, a resync will be needed for each before ticket purchases can continue. If fetching the block info isn't expensive I'd say we keep the check. Thoughts?